### DISCUSSION TITLE
Is it possible to control the placement of an overlay/underlay?

### DISCUSSION THREAD

**Discussion by AlisterH** [qpdf/qpdf#567] [Q&A]
Hi, I'm posting this as a discussion rather than raising it again on a closed ticket, and I'm hesitant to create a new issue as I'm not certain that it isn't currently possible.

The overlay/underlay feature is great, but unless I'm missing something, there is no control on how the overlay or underlay is placed on the page - it is simply placed in the centre, and scaled down if necessary to fit.  Is this right?  Or is there some way of controlling it that I'm missing?

As I described at https://github.com/qpdf/qpdf/issues/207#issuecomment-517494857, I imagine a very common use case is wanting to place a stamp in a corner of the pdf. If all pages are the same size and orientation this is easily accomplished by creating a full page stamp. If the pages are all the same shape and orientation I guess you can just create the stamp at the largest page size, and it will always be placed in the same corner (although it may not be desirable that it is scaled down on smaller pages).
But if some pages are portrait and others are landscape, a portrait stamp will come out wrong on a landscape page, and vice versa. See screenshot illustrating the results of attempting to apply the same stamps to both portrait and landscape pages.
![image](https://user-images.githubusercontent.com/714600/62334010-08078b00-b51a-11e9-880c-409850340c89.png)

Thanks.

**Comment by jberkenbilt** [comment]
As I prepare to break ground on what I'm calling the "pages" epic, I'm creating an issue from this and tagging it since this is in scope for that work.

**Comment by jberkenbilt** [comment]
Closing discussion in favor of #1031
